crates/fuel-streams-core/src/stream.rs

```diff
@@ -106,27 +104,36 @@ impl<S: Streamable> Stream<S> {
         &self,
         subject: &impl IntoSubject,
         payload: &S,
-    ) -> Result<u64, StreamError> {
+    ) -> Result<i64, StreamError> {
```

Comment:
Since we don't always have the sequence value, especially in the case when it has been published, it might be better to simplify the interface:

```suggestion
    ) -> Result<(), StreamError> {
```

Reply:
agree. Done.